server/src/__tests__/donatedItem.test.ts

```diff
@@ -0,0 +1,72 @@
+import request from 'supertest';
+import express from 'express';
+import donatedItemRoutes from '../routes/donatedItemRoutes';
+import mockPrismaClient from '../__mocks__/mockPrismaClient';
+import { newProgram } from '../__mocks__/mockProgram';
+import { newDonor } from '../__mocks__/mockDonor';
+import { newItem } from '../__mocks__/mockNewItem';
+import { updateData } from '../__mocks__/mockUpdateData';
+
+const app = express();
+app.use(express.json());
+app.use('/donatedItem', donatedItemRoutes);
+
+describe('DonatedItem API Tests', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        // Setup mock responses
+        mockPrismaClient.program.findUnique.mockResolvedValue(newProgram);
+        mockPrismaClient.donor.findUnique.mockResolvedValue(newDonor);
+        mockPrismaClient.donatedItem.create.mockResolvedValue({ id: 1, ...newItem });
+        mockPrismaClient.donatedItem.update.mockResolvedValue({ id: 1, ...updateData });
+    });
+
+    // Test POST /donatedItem
+    it('validates Program name and Donor email before creating a donated item', async () => {
+        const response = await request(app).post('/donatedItem').send(newItem);
+        expect(response.status).toBe(201);
+        expect(mockPrismaClient.donatedItem.create).toHaveBeenCalled();
+    
+    });
+
+    it('handles errors when the provided Program or Donor does not exist', async () => {
```

Comment:
@truffer11 It wouldn't check in the real or actual database; it will use mock prisma client and you can see at [this line](https://github.com/oss-slu/material-donor-mutual-assist/pull/49/files#diff-cacffbaf92d007aa8421f8e53bdff31e2a3a00d9553542b6b548764b82a7d823R35) that it resolves to value `null`  and it always resolves to null values irrespective of the size of database